PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx

```diff
@@ -1013,36 +1013,52 @@ void AliAnalysisTaskSEDs::UserExec(Option_t * /*option*/)
 
         if (fApplyML)
         {
-          isMLsel = fMLResponse->IsSelected(modelPred, d, aod->GetMagneticField(), Pid_HF, 0);
+            isMLsel = fMLResponse->IsSelectedMultiClass(modelPred, d, aod->GetMagneticField(), Pid_HF);
+
+          if (isMLsel)
+          {
+            if (modelPred.size() == 0)
+            {
+              modelPred.push_back(-1.);
+              modelPred.push_back(-1.);
+              modelPred.push_back(-1.);
+            }
+            else if (modelPred.size() == 1)
+            {
+              modelPred.push_back(-1.);
+              modelPred.push_back(-1.);
+            }
+          }
 
           if (fEnablePIDMLSparses && (!fReadMC || (indexMCKKpi == GetSignalHistoIndex(iPtBin) && orig == 4)))
           {
-            Double_t var4nSparsePID[knVarPID] = {ptCand, modelPred,
-                                                fMLResponse->GetVariable("nsigTPC_Pi_0"), fMLResponse->GetVariable("nsigTPC_K_0"),
-                                                fMLResponse->GetVariable("nsigTOF_Pi_0"), fMLResponse->GetVariable("nsigTOF_K_0"),
-                                                fMLResponse->GetVariable("nsigTPC_Pi_1"), fMLResponse->GetVariable("nsigTPC_K_1"),
-                                                fMLResponse->GetVariable("nsigTOF_Pi_1"), fMLResponse->GetVariable("nsigTOF_K_1"),
-                                                fMLResponse->GetVariable("nsigTPC_Pi_2"), fMLResponse->GetVariable("nsigTPC_K_2"),
-                                                fMLResponse->GetVariable("nsigTOF_Pi_2"), fMLResponse->GetVariable("nsigTOF_K_2")};
-
-            Double_t var4nSparsePIDcomb[knVarPIDcomb] = {ptCand, modelPred,
+            std::vector<Double_t> var4nSparsePID = {ptCand, modelPred[0], modelPred[1], modelPred[2],
```

Comment:
See comment in the `CreateCutVarsAndEffSparses` method, but it would be convenient to move the ML outputs at the end of this vector, so that if it is binary classification you can simply reduce by 2 the number of axes